File: Interview Questions/Prime.java
public class Prime {
    static boolean isPrime(double x) {
	for (double i = 2; i < Math.floor(Math.sqrt(x)) + 1; i++) {
	    if ((x % i) == 0) {
	       	return false;
	    }
	}	
	return true;
    }

    public static void main(String args[]) {
	double num = 600851475143.0;
	for (double i = Math.floor(Math.sqrt(num)); i >= 1; i--) {
	    if ((num % i) == 0 && isPrime(i)) {
		System.out.println(i);
		break;
	    }
	}
    }
}

File: Interview Questions/prime.cpp
#include <iostream>
#include <cmath>

using natural = unsigned long;
bool isPrime(natural n) {
    for (natural i = 2; i < sqrt(n) + 1; i++) {
	if (fmod(n,i) == 0) {
	    return false;
	}
    }
    return true;
}

int main() {
    natural num = 600851475143L;
    for (natural i = sqrt(num)+1; i >= 1; i--) {
	if (fmod(num,i) == 0 && isPrime(i)) {
	    std::cout << i << std::endl;
	    break;
	}
    }
    return 0;
}
